addons/crm/tests/test_crm_pls.py

```diff
@@ -164,9 +164,10 @@ def test_predictive_lead_scoring(self):
             self._get_lead_values(False, 'no_team_%s' % str(13), country_ids[0], state_ids[1], state_values[2], state_values[0], source_ids[2], stage_ids[1]))
 
         leads = Lead.create(leads_to_create)
+        Lead.sudo().with_context(active_test=False).search([('id', 'not in', leads.ids)]).unlink()
 
-        # assign team 3 to all leads with no teams (also take data into account).
-        leads_with_no_team = self.env['crm.lead'].sudo().search([('team_id', '=', False)])
+        # assign team 3 to all leads with no teams
+        leads_with_no_team = leads.sudo().filtered(lambda lead: not lead.team_id)
```

Comment:
Note that due to the unlink above, this is probably not necessary anymore (but we could assert that leads_with_no_team == leads.filtered(), ensuring there is no data interfering in the test)